### ISSUE TITLE
Symfony Response on 416 HTTP Exception

### ISSUE THREAD

**Issue by DmitryKmita** [symfony/symfony#13465]
Version of symfony: 2.6.3
Version of cUrl: 7.19.7

If I send a request to an action that throws 416 Exception the curl body is empty.
In ~2.5 version of the symfony there was no such problem.

Because of this problem if you have a test that validates the error page and uses cUrl to parse the DOM- it will get an error "The current node list is empty".

By specs in 416 HTTP error response should have:
Content-Range header.
I compared the response between ~2.5 version and 2.6.3 and found that in 2.6.3 I don't get
Transfer-Encoding: chunked header, maybe this is the reason why response body is empty.

**Comment by jaytaph** [comment]
There was recently a patch that dealt with adding an invalid Content-Range header on non-safe request (like POST, PUT etc).

Are you fetching the page through a GET request?

**Comment by jakzal** [comment]
@DmitryKmita would you mind to provide code to reproduce your issue?

**Comment by DmitryKmita** [comment]
This is a PUT request.
Code in symfony action:
```
throw new HttpException(416, "some message");
```
and make simple CURL request from CLI to this action:
```
curl -X GET '%URL%' -v
```

**Comment by jaytaph** [comment]
>By specs in 416 HTTP error response should have: `Content-Range` header.
Not quite, the spec (http://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html) says that a `Content-Range` SHOULD be added when a range-request has been made (ie: when you have specified something like: `Content-Range: bytes 10-20` in your request). Basically telling it: i could not satisfy returning a response for bytes 10-20.

If you did not specify a content-range, but added an `Accept-Range: meters`, where `meters` is a range-format you want to use (http://tools.ietf.org/html/rfc7233#section-2.2), or you just add `Accept-Range: bytes), and you don't support ranged responses, then you can send a 416 back without a content-range, telling the client that you are not able to return a ranged response in the requested format.

`Transfer-encoding: Chunked` has nothing to do with ranges, it's just that the response output is of indefinite size (like passing output from a shell command, or database rows), so it will not send a `content-length` header. It will keep on pushing response data and return an empty chunk at the end, marking the end of the response. 

I can check to see if I can reproduce this on both Symfony versions though.

**Comment by jakzal** [comment]
> By specs in 416 HTTP error response should have: Content-Range header.

This has been recently fixed by #18080

**Comment by jakzal** [comment]
Closing as I can't reproduce getting an empty response when throwing the `HttpException` (neither on 2.3 nor 2.7). In dev environment an exception page is shown, while in prod the standard error page.
